truffle-config.js

```diff
@@ -1,119 +1,66 @@
-    // if it's defined here and no other network is specified at the command line.
-    // You should run a client (like ganache-cli, geth or parity) in a separate terminal
-    // tab if you use this network and you must also set the `host`, `port` and `network_id`
-    // options below to some value.
-    //
-    development: {
-      host: "127.0.0.1",     // Localhost (default: none)
-      port: 8545,            // Standard Ethereum port (default: none)
-      network_id: "*",       // Any network (default: none)
-    },
-    coverage: {
-      host: "localhost",
-      network_id: "*",
-      port: 8555,         // <-- If you change this, also set the port option in .solcover.js.
-      gas: 0xfffffffffff, // <-- Use this high gas value
-      gasPrice: 0x01      // <-- Use this low gas price
-    },
+// Gas limit
+const gas = process.env.GAS_LIMIT || DEFAULT_GAS_LIMIT
 
-    // Another network with more advanced options...
-    // advanced: {
-    // port: 8777,             // Custom port
-    // network_id: 1342,       // Custom network
-    // gas: 8500000,           // Gas sent with each transaction (default: ~6700000)
-    // gasPrice: 20000000000,  // 20 gwei (in wei) (default: 100 gwei)
-    // from: <address>,        // Account to send txs from (default: accounts[0])
-    // websockets: true        // Enable EventEmitter interface for web3 (default: false)
-    // },
+// Allow to add an additional network (useful for docker-compose setups)
+//  i.e. NETWORK='{ "name": "docker", "networkId": "99999", "url": "http://rpc:8545", "gas": "6700000", "gasPrice": "25000000000"  }'
+let additionalNetwork = process.env.NETWORK ? JSON.parse(process.env.NETWORK) : null
 
-    // Useful for deploying to a public network.
-    // NB: It's important to wrap the provider as a function.
-    // ropsten: {
-    // provider: () => new HDWalletProvider(mnemonic, `https://ropsten.infura.io/${infuraKey}`),
-    // network_id: 3,       // Ropsten's id
-    // gas: 5500000,        // Ropsten has a lower block limit than mainnet
-    // confirmations: 2,    // # of confs to wait between deployments. (default: 0)
-    // timeoutBlocks: 200,  // # of blocks before a deployment times out  (minimum/default: 50)
-    // skipDryRun: true     // Skip dry run before migrations? (default: false for public nets )
-    // },
 
-    // Useful for private networks
-    // private: {
-    // provider: () => new HDWalletProvider(mnemonic, `https://network.io`),
-    // network_id: 2111,   // This network is yours, in the cloud.
-    // production: true    // Treats this network as if it was a public net. (default: false)
-    // }
-  },
+const {
+  gas: gasLog,
+} = require('minimist')(process.argv.slice(2), { alias: { gas: 'g' } })
+
 
-  // Set default mocha options here, use special reporters etc.
+module.exports = {
+  ...truffleConfig({
+    mnemonic,
+    privateKey,
+    urlRinkeby: 'https://rinkeby.infura.io/v3/9408f47dedf04716a03ef994182cf150',
```

Comment:
This string at the end is not an API key, is it?

Reply:
Yep, it's one that we are ok sharing. If they overuse it the api key we can invalidate it. We use another one locally. 
If you think it's too bad we can remove it, but I think there's plenty of for example front ends that use infura directly and you only need to see the network request to see it

I see it ok if we provide a default like:
```js
const infuraApiKey = process.env.INFURA_API_KEY || '9408f47dedf04716a03ef994182cf150'
```

The point is to allow people to download and use removing the steps of generating an API key and add it to their env file